pkg/controller/nstemplateset/nstemplateset_controller_test.go

```diff
@@ -306,6 +306,7 @@ func TestReconcileProvisionOK(t *testing.T) {
 			assert.Equal(t, reconcile.Result{}, res)
 			AssertThatNSTemplateSet(t, namespaceName, username, fakeClient).
 				HasFinalizer().
+				HasLabel(toolchainv1alpha1.ProviderLabelKey, toolchainv1alpha1.ProviderLabelValue).
```

Comment:
The NSTemplateSet is not supposed to have the label. You should verify it in the namespace.
```suggestion
```